pulsar-connectors/core/src/main/java/org/apache/pulsar/connect/runtime/SinkConnectorRunner.java

```diff
@@ -0,0 +1,163 @@
+            final String subscription = getProperty(ConnectorConfiguration.KEY_SUBSCRIPTION);
+            final ConsumerConfiguration configuration =
+                    new ConsumerConfiguration().setSubscriptionType(SubscriptionType.Failover);
+
+            // create a subscription and start processing messages
+            try (Consumer consumer = client.subscribe(topic, subscription, configuration)) {
+                LOG.info("Running sink connector {} for topic {} and subscription {}",
+                        connector.getClass().getSimpleName(), topic, subscription);
+                LOG.info("acknowledgement interval {} Mb", Bytes.toMb(commitIntervalBytesMb));
+
+                runSinkConnector(consumer);
+
+            } catch (PulsarClientException pce) {
+                LOG.info("unable to create subscribe to topic {} "
+                        + "with subscription {}", topic, subscription);
+                throw new ConnectorExecutionException(pce);
+            }
+        } catch (PulsarClientException pce) {
+            throw new ConnectorExecutionException(pce);
+        } finally {
+            connector.close();
+        }
+    }
+
+    private void runSinkConnector(Consumer consumer) {
+        long bytesProcessed = 0;
+        final List<MessageId> messageIds = new ArrayList<>();
+        Message currentMessage;
+        while (keepGoing) {
+            try {
+                currentMessage = consumer.receive();
+                messageIds.add(currentMessage.getMessageId());
+
+                connector.processMessage(currentMessage);
+                bytesProcessed += currentMessage.getData().length;
+
+                // TODO add an option for a flush interval
+                // is it time to acknowledge?
+                if (bytesProcessed >= commitIntervalBytesMb) {
+                    connector.commit();
+
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("acknowledging {} messages", messageIds.size());
+                    }
+                    acknowledge(consumer, messageIds);
+                    bytesProcessed = 0;
+
+                    // clear ids since we just acknowledged
+                    messageIds.clear();
+                }
+            } catch (Exception ex) {
+                throw new ConnectorExecutionException(ex);
+            }
+        }
+    }
+
+    private void acknowledge(Consumer consumer, List<MessageId> messageIds)
+            throws PulsarClientException {
+        // TODO wait for these to finish
+        for (MessageId messageId : messageIds) {
+            consumer.acknowledgeAsync(messageId);
+        }
+    }
+
+    private String getProperty(String key) {
+        return properties.getProperty(key);
+    }
+
+    private PulsarClient createClient(Properties properties) throws PulsarClientException {
+        return PulsarUtils.createClient(properties);
+    }
+
+    public static SinkConnectorRunner fromProperties(Properties properties) {
```

Comment:
I will update the ConnectorConfiguration and then pass it to the runner instead of a Properties class.